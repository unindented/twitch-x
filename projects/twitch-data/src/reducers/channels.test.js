import reducer from './channels'

const searchChannelsFixture = require('../__fixtures__/searchChannels.json')

describe('reducers/channels', () => {
  describe('default', () => {
    it('returns initial state', () => {
      const prevState = undefined
      expect(reducer(prevState)).toMatchSnapshot()
    })
  })

  describe('LOAD_SEARCH_CHANNELS_SUCCESS', () => {
    it('stores the channels', () => {
      const prevState = {}
      const action = {type: 'LOAD_SEARCH_CHANNELS_SUCCESS', payload: {response: searchChannelsFixture}}
      expect(reducer(prevState, action)).toMatchSnapshot()
    })
  })
})
