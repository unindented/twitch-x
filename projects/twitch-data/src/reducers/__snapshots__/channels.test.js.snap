// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/channels LOAD_SEARCH_CHANNELS_SUCCESS stores the channels 1`] = `
Object {
  "byId": Object {
    "day9": Object {
      "game": null,
      "id": "day9",
      "image": null,
      "name": "day9",
      "status": null,
    },
    "day9daily": Object {
      "game": null,
      "id": "day9daily",
      "image": "https://static-cdn.jtvnw.net/jtv_user_pictures/day9daily-profile_image-14e040a6b9a079be-300x300.jpeg",
      "name": "day9daily",
      "status": "LIVE from my iPhone 4!",
    },
    "day9tv": Object {
      "game": "Dota 2",
      "id": "day9tv",
      "image": "https://static-cdn.jtvnw.net/jtv_user_pictures/day9tv-profile_image-2226c4b27aaf487a-300x300.png",
      "name": "Day9tv",
      "status": "10v10 DayKnight Festival!  Play Games w/ us!  discord.gg/day9tv",
    },
    "deaths_day98": Object {
      "game": "Battlefield Hardline",
      "id": "deaths_day98",
      "image": null,
      "name": "Deaths_day98",
      "status": "BFH with friends",
    },
    "dota2ti": Object {
      "name": "dota2ti",
    },
  },
  "search": Array [
    "day9tv",
    "day9",
    "day9daily",
    "deaths_day98",
  ],
}
`;

exports[`reducers/channels default returns initial state 1`] = `
Object {
  "byId": Object {},
  "search": Array [],
}
`;
